pkg/providers/v1/aws_sdk_test.go

```diff
@@ -0,0 +1,163 @@
+package aws
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws/retry"
+	awsConfig "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/stretchr/testify/assert"
+
+	"k8s.io/cloud-provider-aws/pkg/providers/v1/config"
+	"k8s.io/cloud-provider-aws/pkg/providers/v1/iface"
+)
+
+type MockAWSSDKProvider struct {
+	p *awsSDKProvider
+}
+
+// Mocks endpoint resolution and custom retries in the EC2 Client.
+func (m *MockAWSSDKProvider) Compute(ctx context.Context, regionName string, mockRT *MockRoundTripper) (iface.EC2, error) {
```

Comment:
from what i see this function is duplicating the code of the actual sdk provider compute for testing purposes but this means that if we are modifying the actual function we also need to update this and its easy to miss it code reviews. Can we not use the actual function with cloud config overrides?